mllib/src/main/scala/org/apache/spark/mllib/feature/HashingTF.scala

```diff
@@ -66,8 +83,9 @@ class HashingTF(val numFeatures: Int) extends Serializable {
   def transform(document: Iterable[_]): Vector = {
     val termFrequencies = mutable.HashMap.empty[Int, Double]
     val setTF = if (binary) (i: Int) => 1.0 else (i: Int) => termFrequencies.getOrElse(i, 0.0) + 1.0
+    val hashFunc: Any => Int = if (hashAlgorithm == Murmur3) murmur3Hash else nativeHash
     document.foreach { term =>
-      val i = indexOf(term)
+      val i = Utils.nonNegativeMod(hashFunc(term), numFeatures)
```

Comment:
@jkbradley I understand what you means now. Your PR looks great and I merged it. Thanks!